### PR TITLE
Force default bucket plugin only if necessary

### PR DESCRIPTION
**Pull request by leplatrem**
@Natim r?

### CODE REVIEW

File: kinto/__init__.py [hunk 1]
```diff
@@ -42,13 +42,17 @@ def main(global_config, config=None, **settings):
                        version=__version__,
                        default_settings=DEFAULT_SETTINGS)
 
-    config.include('kinto.plugins.default_bucket')
+    settings = config.get_settings()
+
+    # In Kinto API 1.x, a default bucket is available.
+    # Force its inclusion if not specified in settings.
+    if 'kinto.plugins.default_bucket' not in settings['includes']:
+        config.include('kinto.plugins.default_bucket')
```
Comment (almet) [inline kinto/__init__.py]: Should we issue a warning and ask for it to be included in the configuration explicitely now?

**Comment by Natim** [comment]
I don't think people should include it in the config so IMHO we can let it as it is.

**Comment by leplatrem** [comment]
> I don't think people should include it in the config so IMHO we can let it as it is.

But that's the only way to have default bucket work with kinto-attachment ! And that's the main reason  we put the default bucket as a plugin :) See #277 

> Should we issue a warning and ask for it to be included in the configuration explicitely now?

yes, I think so

**Comment by almet** [comment]
Then, how do you remove the feature?

**Comment by Natim** [comment]
As far as I know we don't aim to make it an optional feature do we?

**Comment by leplatrem** [comment]
> Then, how do you remove the feature?

We cannot at this stage.

Currently the client of v1 expects the default bucket to create collections implicitly.

Being able to remove the default bucket could be part of the API v2.

**Comment by leplatrem** [comment]
Ready to review/merge!

File: kinto/config/kinto.tpl [hunk 2]
```diff
@@ -36,6 +36,11 @@ multiauth.policies = basicauth
 # kinto.experimental_collection_schema_validation = true
 
 #
+# Plugins
+#
+kinto.includes = kinto.plugins.default_bucket
```
Comment (Natim) [inline kinto/config/kinto.tpl]: It seems strange to me to add it there if it is already added automatically.

Comment (leplatrem) [inline kinto/config/kinto.tpl]: So that adding it automatically could be removed in 2.X, whereas config files can stay

**Comment by Natim** [comment]
r+

**Pull request merged**